controllers/ingress/group_controller.go

```diff
@@ -43,11 +44,21 @@ const (
 )
 
 // NewGroupReconciler constructs new GroupReconciler
-func NewGroupReconciler(cloud aws.Cloud, k8sClient client.Client, eventRecorder record.EventRecorder,
-	finalizerManager k8s.FinalizerManager, networkingSGManager networkingpkg.SecurityGroupManager,
-	networkingSGReconciler networkingpkg.SecurityGroupReconciler, subnetsResolver networkingpkg.SubnetsResolver,
-	elbv2TaggingManager elbv2deploy.TaggingManager, controllerConfig config.ControllerConfig, backendSGProvider networkingpkg.BackendSGProvider,
-	sgResolver networkingpkg.SecurityGroupResolver, logger logr.Logger) *groupReconciler {
+func NewGroupReconciler(
+	cloud aws.Cloud,
+	k8sClient client.Client,
+	eventRecorder record.EventRecorder,
+	finalizerManager k8s.FinalizerManager,
+	networkingSGManager networkingpkg.SecurityGroupManager,
+	networkingSGReconciler networkingpkg.SecurityGroupReconciler,
+	subnetsResolver networkingpkg.SubnetsResolver,
+	elbv2TaggingManager elbv2deploy.TaggingManager,
+	targetGroupMetric *prometheus.GaugeVec,
```

Comment:
nit: maybe name it as `targetGroupMetricAgent` or  `targetGroupMetricCollect` ?

Reply:
I don't see how that would make the code clearer